runtime/queries/vhdl/highlights.scm

```diff
@@ -1,338 +1,134 @@
+  "for" "loop" "while"
+] @keyword.repeat
+
+[ 
+  "if" "elsif" "else" "case" "then" "when"
+] @keyword.conditional
+
+[ 
+  "library" "use"
+] @keyword.include
+
+(comment) @comment @spell
+
+(type_mark) @type
 
 [
-  ; vhdl 02
-  "base"
-  "left"
-  "right"
-  "high"
-  "low"
-  "pos"
-  "val"
-  "succ"
-  "pred"
-  "leftof"
-  "rightof"
-  "range"
-  "reverse_range"
-  "length"
-  "delayed"
-  "stable"
-  "quiet"
-  "transaction"
-  "event"
-  "active"
-  "last_event"
-  "last_active"
-  "last_value"
-  "driving"
-  "driving_value"
-  "ascending"
-  "value"
-  "image"
-  "simple_name"
-  "instance_name"
-  "path_name"
-  ;"foreign"
-  ; vhdl 08
-  "instance_name"
-  "path_name"
-] @attribute
+  "(" ")" "[" "]"
+] @punctuation.bracket
 
-;[
-  ; vhdl 02
-  ;"now"
-  ;"resolved"
-  ;"rising_edge"
-  ;"falling_edge"
-  ;"read"
-  ;"readline"
-  ;"hread"
-  ;"oread"
-  ;"write"
-  ;"writeline"
-  ;"hwrite"
-  ;"owrite"
-  ;"endfile"
-  ;"resize"
-  ;"is_X"
-  ;"std_match"
-  ;"shift_left"
-  ;"shift_right"
-  ;"rotate_left"
-  ;"rotate_right"
-  ;"to_unsigned"
-  ;"to_signed"
-  ;"to_integer"
-  ;"to_stdLogicVector"
-  ;"to_stdULogic"
-  ;"to_stdULogicVector"
-  ;"to_bit"
-  ;"to_bitVector"
-  ;"to_X01"
-  ;"to_X01Z"
-  ;"to_UX01"
-  ;"to_01"
-  ;"conv_unsigned"
-  ;"conv_signed"
-  ;"conv_integer"
-  ;"conv_std_logic_vector"
-  ;"shl"
-  ;"shr"
-  ;"ext"
-  ;"sxt"
-  ;"deallocate"
-  ; vhdl 08
-  ;"finish"
-  ;"flush"
-  ;"justify"
-  ;"maximum"
-  ;"minimum"
-  ;"resolution_limit"
-  ;"stop"
-  ;"swrite"
-  ;"tee"
-  ;"to_binarystring"
-  ;"to_bstring"
-  ;"to_hexstring"
-  ;"to_hstring"
-  ;"to_octalstring"
-  ;"to_ostring"
-  ;"to_string"
-  ; vhdl math
-  ;"arccos"
-  ;"arccosh"
-  ;"arcsin"
-  ;"arcsinh"
-  ;"arctan"
-  ;"arctanh"
-  ;"arg"
-  ;"cbrt"
-  ;"ceil"
-  ;"cmplx"
-  ;"complex_to_polar"
-  ;"conj"
-  ;"cos"
-  ;"cosh"
-  ;"exp"
-  ;"floor"
-  ;"get_principal_value"
-  ;"log"
-  ;"log10"
-  ;"log2"
-  ;"polar_to_complex"
-  ;"realmax"
-  ;"realmin"
-  ;"round"
-  ;"sign"
-  ;"sin"
-  ;"sinh"
-  ;"sqrt"
-  ;"tan"
-  ;"tanh"
-  ;"trunc"
-  ;"uniform"
-;] @function.builtin
+[
+  "." ";" "," ":"
+] @punctuation.delimeter
 
-; Operators
 [
-  "+"
-  "-"
-  "*"
-  "/"
-  "**"
-  "abs"
-  "not"
-  "mod"
-  "rem"
-  "&"
-  "sll"
-  "srl"
-  "sla"
-  "sra"
-  "rol"
-  "ror"
-  "="
-  "/="
-  "?="
-  "?/="
-  "?<"
-  "?<="
-  "?>"
-  "?>="
-  "<"
-  "<="
-  ">"
-  ">="
-  "and"
-  "or"
-  "nand"
-  "nor"
-  "xor"
-  "xnor"
-  ":="
-  "<="
-  "??"
+  "=>" "<=" "+" ":=" "=" "/=" "<" ">" "-" "*"
+  "**" "/" "?>" "?<" "?<=" "?>=" "?=" "?/="
+; "?/" errors, maybe due to escape character
+  (attribute_name "'")
+  (index_subtype_definition (any))
 ] @operator
 
 [
-  ";"
-  ","
-] @punctuation.delimiter
+  "not" "xor" "xnor" "and" "nand" "or" "nor" "mod" "rem"
+  (attribute_name "'")
+  (index_subtype_definition (any))
+] @keyword.operator
 
 [
-  "("
-  ")"
-  "'"
-] @punctuation.bracket
+  (real_decimal)
+  (integer_decimal)
+] @number
```

Comment:
The default theme isn't styling everything because wrong scope names are used. https://docs.helix-editor.com/themes.html#syntax-highlighting

`@constant.numeric`

Reply:
> The default theme isn't styling everything because wrong scope names are used. https://docs.helix-editor.com/themes.html#syntax-highlighting
> 
> `@constant.numeric`

That makes sense! I wish I had someone help me out when I initially did the highlights, might've gotten it a lot closer.

Reply:
Done. Thanks for the link. As it turns out, reading the friendly manual helps. Fyi I got the scope names from nvim-treesitter on their Verilog highlights.scm